Goodpratices, Technical Tools and New Technical Keys Steps (#260)
PrestaShop/docs: src/content/1.7/modules/creation/good-practices.md

- Prefix what belongs to you:
  - modules
  - configuration parameters (ps_configuration)
  - Smarty variables
  - tables
  - CSS classes
  
- Consider deleting your configurations, tables, admin tabs, and all other entities unique to your product during the uninstallation of your modules.

- You shouldn't use HTML code in your PHP code. Here are alternatives to separate the view into your code:
  - display content with [Smarty][display-content-front-office]
  - use [helpers][adding-configuration-page] for your configuration pages
  - include [PrestUI](https://github.com/Scritik/prestui), a graphic library created by the community for your configuration pages.

> - We recommend to create your module's interface so it matches [PrestaShop's UI kit](http://build.prestashop.com/prestashop-ui-kit/).

Reviewer: 👍

Author: I was told throught slack to not communicate that link.
So which one ?

- For modules compatible before the 1.7 PrestaShop version, the minimal compatibility has to be PHP 5.3. You will be then able to use, for example, namespaces.

> - If your module has the possibility to upload a file, check the type of the file to make sure you avoid security issues. Here are more information for the function [mime_content_type()](http://php.net/manual/en/function.mime-content-type.php).

Reviewer 2: ```suggestion
- If your module allows the upload of a file, check the type of the file to make sure you avoid security issues. Here are more information for the function [mime_content_type()](http://php.net/manual/en/function.mime-content-type.php).
```

- When your module has forms, you should:

>   - show a confirmation message if everything went fine or an error message if not.

Reviewer 2: ```suggestion
  - show a confirmation message if everything went fine or an error message if it did not.
```

  - make sure information entered by customers are correct. If you ask a sum, it has to be only numbers. More information about the Validate class of PrestaShop [here](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Validate.php).

- Consider carefully casting your variables and use pSQL/bqSQL in the SQL requests to avoid any injections (read [Best Practices of the Db Class](http://doc.prestashop.com/display/PS16/Best+Practices+of+the+Db+Class)). Make sure your files are properly protected (especially if your module uses a cron for example) to avoid anyone being able to execute them. As a result, you are required to use a token!

- The use of overrides is permitted, however if we decide that too many (2 / 3 max) have been used and/or the modifications are too dangerous, we will refuse your module. If you're unsure, don't hesitate to get in touch.

> - If you need to load a .js or .tpl (containing JavaScript) to the back-office, put restrictions in place on your hooks (hookbackofficeheader/top/footer)! Specific examples: If your JS only applies to your module's configuration page, use:  

Reviewer 2: We need to harmonize the wording, there are currently backoffice, back office, back-office... it needs to be back office. Can you please modify it? Thank you.

> - If you need to load a .js or .tpl (containing JavaScript) to the back-office, put restrictions in place on your hooks (hookbackofficeheader/top/footer)! Specific examples: If your JS only applies to your module's configuration page, use:  

Reviewer 2: ```suggestion
- If you need to load a .js or .tpl (containing JavaScript) to the back-office, put restrictions in place on your hooks (hookbackofficeheader/top/footer)! Specific examples: if your JS only applies to your module's configuration page, use:  
```
